server/py/framework/db/sqldb/db.py

```diff
@@ -6943,12 +6937,10 @@ def _is_table_exists(session: Session, table_name: str) -> bool:
 
     @staticmethod
     def _paginate_query(
-        query, page: typing.Optional[int] = None, page_size: typing.Optional[int] = None
+        query, offset: typing.Optional[int] = None, limit: typing.Optional[int] = None
     ):
-        if page is not None:
-            page_size = page_size or config.httpdb.pagination.default_page_size
-            if query.count() < page_size * (page - 1):
-                raise StopIteration
-            query = query.limit(page_size).offset((page - 1) * page_size)
-
+        if offset:
+            query = query.offset(offset)
+        if limit:
+            query = query.limit(limit)
```

Comment:
nitpick
if limit is 0, then you essentially return all results

I suggest to check None type